include/crm/common/xml.h

```diff
@@ -312,6 +312,48 @@ int xml_apply_patchset(xmlNode *xml, xmlNode *patchset, bool check_version);
 
 void patchset_process_digest(xmlNode *patch, xmlNode *source, xmlNode *target, bool with_digest);
 
+/* exclusive (should any combination make sense -> explicitly enumerated),
+   with intentional symbolic constant duality (easier to adapt to growth) */
+enum pcmk_acl_cred_type {
+#define PCMK_ACL_CRED_USER PCMK_ACL_CRED_USER
```

Comment:
I don't understand what this definition accomplishes

Reply:
My view radically differs, this is highly practical for simple projects
for which actually any bloated build system (like said `configure` checks)
is a clutter on it's own, when they can just `#ifdef` in code as they see
fit -- conclusion shall not be jumped to based on a single project.

Calling 3 extra lines a clutter?  This is the same approach used
in the standard library, one would be mad translating any ad-hoc
`#ifdef`s in code into `configure` checks + consequently another
`#ifdef`s based on the result of those `configure` checks (doesn't
it resemble the notion of "clutter"?).

Also note that `configure` checks can never be as up-to-date as `#ifdef`
when actually compiling (check-during-build is sort of atomic and the
only _static_ source of truth, and regarding _static_ limitations, that's
why there's this further _dynamic_ indicated-unsupportability-by-return-code
rectification when the two accidentally differ).